packages/edit-site/src/components/block-preview/index.js

```
@@ -0,0 +1,167 @@
+				minHeight,
+			} }
+		>
+			<Iframe
+				contentRef={ useRefEffect( ( bodyElement ) => {
+					const {
+						ownerDocument: { documentElement },
+					} = bodyElement;
+					documentElement.classList.add(
+						'block-editor-block-preview__content-iframe'
+					);
+					documentElement.style.position = 'absolute';
+					documentElement.style.width = '100%';
+
+					// Necessary for contentResizeListener to work.
+					bodyElement.style.boxSizing = 'border-box';
+					bodyElement.style.position = 'absolute';
+					bodyElement.style.width = '100%';
+				}, [] ) }
+				aria-hidden
+				tabIndex={ -1 }
+				style={ {
+					position: 'absolute',
+					width: viewportWidth,
+					height: contentHeight,
+					pointerEvents: 'none',
+					// This is a catch-all max-height for patterns.
+					// See: https://github.com/WordPress/gutenberg/pull/38175.
+					maxHeight: MAX_HEIGHT,
+					minHeight:
+						scale !== 0 && scale < 1 && minHeight
+							? minHeight / scale
+							: minHeight,
+				} }
+			>
+				<EditorStyles styles={ editorStyles } />
+				{ contentResizeListener }
+				<div
+					dangerouslySetInnerHTML={ {
+						__html: html,
+					} }
+				></div>
+			</Iframe>
+		</Disabled>
+	);
+}
+
+export default function BlockPreview( props ) {
+	const { blocks } = props;
+	const [ containerResizeListener, { width: containerWidth } ] =
+		useResizeObserver();
+
+	const [ html, setHTML ] = useState( '' );
+
+	const { editedPostId } = useSelect( ( select ) => {
+		const { getEditedPostId } = select( editSiteStore );
+		return {
+			editedPostId: getEditedPostId(),
+		};
+	}, [] );
+
+	useEffect( () => {
+		const getHTML = async () => {
+			const dataHTML = await apiFetch( {
+				path: '/wp/v2/render_blocks',
+				method: 'POST',
+				data: {
+					blocks: serialize( blocks ),
+					post_id: editedPostId,
+				},
+			} );
+			setHTML( dataHTML );
+		};
+		getHTML().catch( ( error ) => {
+			return error;
```

What's happening with this error? It seems to be swallowed right?

Should we instead set some local error state (as you do with the `setHTML`) and then use this to display an error message instead of the preview? Or you could revert to rendering client side.